src/transformers/generation/logits_process.py

```diff
@@ -137,22 +139,24 @@ class MinLengthLogitsProcessor(LogitsProcessor):
     ```
     """
 
-    def __init__(self, min_length: int, eos_token_id: Union[int, List[int], torch.Tensor]):
+    def __init__(self, min_length: int, eos_token_id: Union[int, List[int], torch.Tensor], device: str = None):
         if not isinstance(min_length, int) or min_length < 0:
             raise ValueError(f"`min_length` has to be a non-negative integer, but is {min_length}")
 
+        if device is None:
+            device = "cpu"
+
```

Comment:
I'd argue that we can just set it to "cpu" in the arg no?

Reply:
This is mostly for users who use/pass LogitsProcessor as a standalone kwarg, because 'generate()' takes care that device is not None.

I think we should raise warning for BC saying users to pass-in the device, but let's ask @gante if he's okay with it. If I am not misunderstanding, we shouldn't raise warnings 🤔 

> Let's not throw a warning when the device is not passed and tensors are initialized on CPU. A .to operation is not that expensive :)

Reply:
yeah, don't think it's a problem to silently do this

Reply:
Down to just default to CPU which was already the behaviour by default before this PR no?

Reply:
ahh my bad, didn't read carefully the first comment. Setting in the arg as default is better, right 

My concern is that before this PR, we were placing these on `scores.device` during "_ [call]_ " , but anyway I still get lost at when to do BC deprecation and when to not do 😄 https://github.com/huggingface/transformers/blob/03935d300d60110bb86edb49d2315089cfb19789/src/transformers/generation/logits_process.py#L155